net/iputils/Makefile

```diff
@@ -0,0 +1,150 @@
+#
+# Copyright (C) 2006-2010 OpenWrt.org
+# Copyright (C) 2019 Noah Meyerhans <frodo@morgul.net>
+#
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=iputils
+PKG_VERSION:=20190709
+PKG_RELEASE:=1
+
+PKG_MAINTAINER:=Noah Meyerhans <frodo@morgul.net>
+PKG_LICENSE:=BSD-3-Clause
+PKG_LICENSE_FILES:=LICENSE
+PKG_CPE_ID:=cpe:/a:iputils_project:iputils
+
+PKG_SOURCE:=iputils-s$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/iputils/iputils/tar.gz/s$(PKG_VERSION)?
+PKG_HASH:=a15720dd741d7538dd2645f9f516d193636ae4300ff7dbc8bfca757bf166490a
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/iputils-s$(PKG_VERSION)
+
+PKG_BUILD_DEPENDS:=meson/host
+include $(INCLUDE_DIR)/package.mk
+include ../../devel/meson/meson.mk
+
+# base meson args that are common across all builds, regardless of
+# other options and independent of specific package selections:
+MESON_ARGS:=-Dprefix=/usr -DBUILD_MANS=false -DBUILD_HTML_MANS=false -DUSE_CAP=false -DUSE_CRYPTO=kernel
+define Build/Configure
+  $(call Build/Configure/Meson)
+endef
+
+define Package/iputils/Default
+  SECTION:=net
+  CATEGORY:=Network
+  TITLE:=iputils
+  URL:=https://github.com/iputils/iputils
+endef
+
+
+### ping
+
+define Package/iputils-ping
+  $(call Package/iputils/Default)
+  TITLE:=iputils-ping
+  MESON_ARGS+=-DBUILD_PING=true
+endef
+
+define Package/iputils-ping/config
+	source "$(SOURCE)/Config_ping.in"
+endef
+
+define Package/iputils-ping/description
+  Send ICMP_ECHOREQUEST packets to IP hosts and report on replies
+endef
+
+define Package/iputils-ping/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_SUID) $(MESON_BUILD_DIR)/ping $(1)/usr/bin/
+ifdef CONFIG_PING_LEGACY_SYMLINKS
+	ln -s ping $(1)/usr/bin/ping4
+	ln -s ping $(1)/usr/bin/ping6
+endif
+endef
+
+### arping
+
+define Package/iputils-arping
+  $(call Package/iputils/Default)
+  TITLE:=iputils-arping
+  MESON_ARGS+=-DBUILD_ARPING=true
+endef
+
+define Package/iputils-arping/description
+  Send ARP REQUEST packets to a neighbor host
+endef
+
+define Package/iputils-arping/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(MESON_BUILD_DIR)/arping $(1)/usr/bin/
+endef
+
+### clockdiff
+
+define Package/iputils-clockdiff
+  $(call Package/iputils/Default)
+  TITLE:=iputils-clockdiff
+  MESON_ARGS+=-DBUILD_CLOCKDIFF=true
```

Comment:
I prefer that these be together. No point in complicating the build like this.

Reply:
I'm not sure what you mean. Are you saying that we should simply build all the binaries unconditionally, but only install the ones that are selected in .config?

Reply:
Yes. I've seen too many build failures where we try to be smart.